### PR TITLE
Improve Convert to Base Nodes

### PR DESCRIPTION
**Pull request by DavidMansolino**
**Description**
Currently, when converting a PROTO node to base nodes all the nodes are converted.
Ideally, only the root node should be converted, the nested PROTO node should remain PROTO nodes (e.g. we don't want to unprototize all the appearance PROTO nodes of a robot PROTO when converting it).

**Related Issues**
Fix issue #1297

**Tasks**
  - [x] Fix conversion.
  - [x] Device whether a 'convertAll' utility would be usefull.
  - [x] Changelog.

### CODE REVIEW

**Comment by DavidMansolino** [comment]
The convert to base node feature is now working as expected, only the selected PROTO node is converted and not the sub-PROTO nodes.
I am wondering if it might still be useful to leave the possibility to the user to convert all the sub-PROTO nodes too (as it is currently on master), because if a user wants to completely 'unprototize' a node it might quickly be a tricky task to track all the internal PROTO nodes (e.g. multiple appearance PROTO nodes).
@cyberbotics/contributors what do you think?

**Comment by stefaniapedrazzi** [comment]
Yes, I think it would be good to still have the possibility convert the whole node to base nodes.
Especially for debugging or modeling, I use it very often and I'm happy that I can do it with a single click without having to care about which are sub-PROTOs.

File: src/webots/user_commands/WbActionManager.cpp [hunk 1]
```diff
@@ -983,10 +983,16 @@ void WbActionManager::populateActions() {
...
   action = new QAction(this);
   action->setText(tr("&Convert to Base Node(s)"));
-  action->setStatusTip(tr("Convert this PROTO node into the equivalent base node(s)."));
+  action->setStatusTip(tr("Convert this PROTO node (and nested PROTO nodes) into the equivalent base node(s)."));
   action->setToolTip(action->statusTip());
   mActions[CONVERT_TO_BASE_NODES] = action;
 
+  action = new QAction(this);
+  action->setText(tr("Convert &Root to Base Node(s)"));
+  action->setStatusTip(tr("Convert this PROTO node into the equivalent base node(s)."));
```
Comment (omichel) [inline src/webots/user_commands/WbActionManager.cpp]:
```suggestion
  action->setStatusTip(tr("Convert this PROTO node into the equivalent base node."));
```

File: src/webots/user_commands/WbActionManager.hpp [hunk 2]
```diff
@@ -138,6 +138,7 @@ class WbActionManager : public QObject {
     SHOW_PROTO_SOURCE,
     SHOW_PROTO_RESULT,
     CONVERT_TO_BASE_NODES,
+    CONVERT_ROOT_TO_BASE_NODES,
```
Comment (omichel) [inline src/webots/user_commands/WbActionManager.hpp]:
```suggestion
    CONVERT_ROOT_TO_BASE_NODE,
```

File: docs/reference/changelog-r2020.md [hunk 3]
```diff
@@ -5,6 +5,7 @@ Released on XXX.
 
   - New Features
     - macOS and Windows: Added support for Python 3.8.
+    - Added a 'Convert Root to Base Node(s)' option in the context menu to convert a PROTO node in base node(s) without converting the nested PROTO nodes.
```
Comment (omichel) [inline docs/reference/changelog-r2020.md]:
```suggestion
    - Added a 'Convert Root to Base Node' option in the context menu to convert a PROTO node to a base node without converting the nested PROTO nodes.
```

File: src/webots/user_commands/WbActionManager.cpp [hunk 4]
```diff
@@ -983,10 +983,16 @@ void WbActionManager::populateActions() {
 
   action = new QAction(this);
   action->setText(tr("&Convert to Base Node(s)"));
-  action->setStatusTip(tr("Convert this PROTO node into the equivalent base node(s)."));
+  action->setStatusTip(tr("Convert this PROTO node (and nested PROTO nodes) into the equivalent base node(s)."));
   action->setToolTip(action->statusTip());
   mActions[CONVERT_TO_BASE_NODES] = action;
 
+  action = new QAction(this);
+  action->setText(tr("Convert &Root to Base Node(s)"));
```
Comment (omichel) [inline src/webots/user_commands/WbActionManager.cpp]:
```suggestion
  action->setText(tr("Convert &Root to Base Node"));
```

**Comment by omichel** [comment]
Please ignore my requested changes from Node(s) to Node as I understood afterwards that several nodes can be created by this operation.

File: docs/reference/changelog-r2020.md (hunk 3 above)
Comment (omichel) [inline docs/reference/changelog-r2020.md]:
```suggestion
    - Added a 'Convert Root to Base Node(s)' option in the context menu to convert a PROTO node to base node(s) without converting the nested PROTO nodes.
```

**Pull request merged**